src/bp/core/routers/admin/versioning.ts

```diff
@@ -1,19 +1,26 @@
 import { Logger } from 'botpress/sdk'
 import { GhostService } from 'core/services'
 import { BotService } from 'core/services/bot-service'
-import { Router } from 'express'
+import { RequestHandler, Router } from 'express'
+import _ from 'lodash'
 
 import { CustomRouter } from '../customRouter'
+import { needPermissions } from '../util'
 
 export class VersioningRouter extends CustomRouter {
-  constructor(logger: Logger, private ghost: GhostService, private botService: BotService) {
+  private needPermissions: (operation: string, resource: string) => RequestHandler
+  private readonly resource = 'admin.versioning'
+
+  constructor(logger: Logger, private ghost: GhostService, private botService: BotService, private workspaceService) {
     super('Versioning', logger, Router({ mergeParams: true }))
+    this.needPermissions = needPermissions(this.workspaceService)
```

Comment:
Maybe we'll have to redefine the roles, because we can't rely on a single user "super admin" to push to prod. I know how it will end up - users are going to share the super admin credentials among themselves so they can circumvent the system.

Reply:
Well, if you are an admin in a workspace, maybe you could just sync the bots in your workspace (locally or remote), and avoid touching global files.

Permissions are useless if a user is allowed to touch global files (clones locally, make changes, then push them)

SuperAdmin: can version global files (& all bots)
Admins: can version his workspace bots

Reply:
I'll revert this to super-admin until we find a better solution